### DISCUSSION TITLE
Optimize get_all_models Function by Adding Caching for Improved Performance

### DISCUSSION THREAD

**Discussion by ssnnow** [open-webui/open-webui#5211] [General]
The get_all_models function is frequently called within our codebase, leading to potential performance bottlenecks. To enhance its efficiency, I propose adding a caching mechanism. By storing previously retrieved models in a cache, we can reduce redundant computations and improve the overall performance of our application.

**Comment by saaldigital-cw** [comment]
I also wanted to add models that could be also disabled (Runpod serverless)
But currently the complete UI is frozen until timeout
